crates/biome_js_analyze/src/assist/source/use_sorted_object_properties.rs

```diff
@@ -0,0 +1,185 @@
+use std::borrow::Cow;
+use std::cmp::Ordering;
+
+use biome_analyze::{context::RuleContext, declare_source_rule, Ast, Rule, RuleAction};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_syntax::{AnyJsObjectMember, AnyJsObjectMemberName, JsObjectMemberList};
+use biome_rowan::{AstSeparatedList, BatchMutationExt, SyntaxResult};
+
+use crate::JsRuleAction;
+
+declare_source_rule! {
+    /// Enforce ordering of a JS object properties.
+    ///
+    /// This rule checks if keys of the object are sorted in a consistent way.
+    /// Keys are sorted in a natural order.
+    /// This rule will consider spread/calculated keys e.g [k]: 1 as non-sortable.
+    /// Instead, whenever it encounters a non-sortable key, it will sort all the
+    /// previous sortable keys up until the nearest non-sortable key, if one
+    /// exist.
+    /// This prevents breaking the override of certain keys using spread
+    /// keys.
+    ///
+    /// Source: https://perfectionist.dev/rules/sort-objects
+    ///
+    /// ## Examples
+    ///
+    /// ```js,expect_diff
+    /// {
+    ///   x: 1,
+    ///   a: 2,
+    /// };
+    /// ```
+    ///
+    /// ```js,expect_diff
+    /// {
+    ///   x: 1,
+    ///   ...f,
+    ///   y: 4,
+    ///   a: 2,
+    ///   [calculated()]: true,
+    ///   b: 3,
+    ///   a: 1,
+    /// };
+    /// ```
+    ///
+    /// ```js,expect_diff
+    /// {
+    ///   get aab() {
+    ///     return this._aab;
+    ///   },
+    ///   set aac(v) {
+    ///     this._aac = v;
+    ///   },
+    ///   w: 1,
+    ///   x: 1,
+    ///   ...g,
+    ///   get aaa() {
+    ///     return "";
+    ///   },
+    ///   u: 1,
+    ///   v: 1,
+    ///   [getProp()]: 2,
+    ///   o: 1,
+    ///   p: 1,
+    ///   q: 1,
+    /// }
+    /// ```
+    pub UseSortedObjectProperties {
+        version: "1.0.0",
```

Comment:
This isn't the version of the rule, it's the version the rule was first released in. This gets updated on release.

```suggestion
        version: "next",
```